hardware/pico_driver/src/pico_driver.cc

```diff
@@ -443,7 +456,7 @@ class PicoDriverL2 : public PicoDriver, public royale::IExtendedDataListener {
     // If we have a listener and the extended data contains intermediate data, publish it
     if (data->hasIntermediateData() && pub_extended_.getNumSubscribers() > 0
         && data->getIntermediateData() != nullptr) {
-      extended_.header.stamp = ros::Time::now();
+      extended_.header.stamp = commonStamp;
```

Comment:
same here

Reply:
The intent of introducing `commonStamp` was to guarantee that the two outgoing messages that are derived from the same sensor frame get identical timestamps.

I mean, hopefully they would have identical timestamps anyway as assigned by the Pico driver, since the acquisition time was the same, but this way it's guaranteed, and you can rely on the timestamps being identical in any downstream analysis algorithms that want to do something with matched pairs of messages.

Reply:
I would understand if the variable `commonStamp` was used elsewhere in the code but it is not, the contents of `data` don't change.

Reply:
My assumptions are that:
- We don't know that `data->getDepthData()->timeStamp` and `data->getIntermediateData()->timeStamp` are the same.
- We don't know in advance which pieces of information (depth, intermediate data) will be defined. It could be either or both.
- In the case that both are defined, we want to guarantee that the two outgoing messages have the same timestamp.

This `commonStamp` solution approach handles it this way:
- If only one piece of information is defined, `commonStamp` gets that timestamp and is used for the one outgoing message.
- If both are defined, `commonStamp` is arbitrarily set to one of them (`data->getDepthData()->timeStamp`) and that is used for both outgoing messages.

I'm not sure I see another way of achieving the design goal that is simpler and equally clear.

Reply:
I was not aware of the design goal "In the case that both are defined, we want to guarantee that the two outgoing messages have the same timestamp" and "If both are defined, commonStamp is arbitrarily set to one of them (data->getDepthData()->timeStamp) and that is used for both outgoing messages."

My assumption was that in the case the timestamps are indeed different, then the desired output timestamp is whatever is more accurate and not necessarily an equal value.